src/ceph-volume/ceph_volume/api/lvm.py

```diff
@@ -863,20 +864,20 @@ def filter(self, lv_name=None, vg_name=None, lv_path=None, lv_uuid=None, lv_tags
             lv_tags={'ceph.osd_id': '0'}
 
         """
-        if not any([lv_name, vg_name, lv_path, lv_uuid, lv_tags]):
+        if not any([dev, lv_name, vg_name, lv_path, lv_uuid, lv_tags]):
             raise TypeError('.filter() requires lv_name, vg_name, lv_path, lv_uuid, or tags (none given)')
-        # first find the filtered volumes with the values in self
-        filtered_volumes = self._filter(
-            lv_name=lv_name,
-            vg_name=vg_name,
-            lv_path=lv_path,
-            lv_uuid=lv_uuid,
-            lv_tags=lv_tags
-        )
-        # then purge everything
-        self._purge()
-        # and add the filtered items
-        self.extend(filtered_volumes)
+
+        if dev is not None:
+            splitname = dmsetup_splitname(dev)
+            if not splitname.get('LV_NAME'):
+                raise DeviceHasNoLVName('device didn\'t have LV_NAME.')
```

Comment:
hmm I'm not a big fan that this function suddenly can throw a custom exception. Wouldn't it be better to move this filtering functionality into `_filter` and simply return an empty list?

Reply:
Ok. I've made the change.